azure/services/agentpools/spec.go

```diff
@@ -58,6 +58,146 @@ type KubeletConfig struct {
+	// Maps to net.ipv4.tcp_keepalive_probes.
+	NetIpv4TCPKeepaliveProbes *int32
+
+	// NetIpv4TCPKeepaliveTime specifies the rate at which TCP sends out a keepalive messages when keepalive is enabled.
+	// Maps to net.ipv4.tcp_keepalive_time.
+	NetIpv4TCPKeepaliveTime *int32
+
+	// NetIpv4TCPMaxSynBacklog specifies the maximum number of queued connection requests that have still not received an acknowledgment from the connecting client.
+	// If this number is exceeded, the kernel will begin dropping requests.
+	// Maps to net.ipv4.tcp_max_syn_backlog.
+	NetIpv4TCPMaxSynBacklog *int32
+
+	// NetIpv4TCPMaxTwBuckets specifies maximal number of timewait sockets held by system simultaneously.
+	// If this number is exceeded, time-wait socket is immediately destroyed and warning is printed.
+	// Maps to net.ipv4.tcp_max_tw_buckets.
+	NetIpv4TCPMaxTwBuckets *int32
+
+	// NetIpv4TCPTwReuse is used to allow to reuse TIME-WAIT sockets for new connections when it's safe from protocol viewpoint.
+	// Maps to net.ipv4.tcp_tw_reuse.
+	NetIpv4TCPTwReuse *bool
+
+	// NetIpv4TCPkeepaliveIntvl specifies the frequency of the probes sent out.
+	// Multiplied by tcpKeepaliveprobes, it makes up the time to kill a connection that isn't responding, after probes started.
+	// Maps to net.ipv4.tcp_keepalive_intvl.
+	NetIpv4TCPkeepaliveIntvl *int32
+
+	// NetNetfilterNfConntrackBuckets specifies the size of hash table used by nf_conntrack module to record the established connection record of the TCP protocol.
+	// Maps to net.netfilter.nf_conntrack_buckets.
+	NetNetfilterNfConntrackBuckets *int32
+
+	// NetNetfilterNfConntrackMax specifies the maximum number of connections supported by the nf_conntrack module or the size of connection tracking table.
+	// Maps to net.netfilter.nf_conntrack_max.
+	NetNetfilterNfConntrackMax *int32
+
+	// VMMaxMapCount specifies the maximum number of memory map areas a process may have.
+	// Maps to vm.max_map_count.
+	VMMaxMapCount *int32
+
+	// VMSwappiness specifies aggressiveness of the kernel in swapping memory pages.
+	// Higher values will increase aggressiveness, lower values decrease the amount of swap.
+	// Maps to vm.swappiness.
+	VMSwappiness *int32
+
+	// VMVfsCachePressure specifies the percentage value that controls tendency of the kernel to reclaim the memory, which is used for caching of directory and inode objects.
+	// Maps to vm.vfs_cache_pressure.
+	VMVfsCachePressure *int32
+}
+
+// LinuxOSConfig specifies the custom Linux OS settings and configurations.
+type LinuxOSConfig struct {
+	// SwapFileSizeMB specifies size in MB of a swap file will be created on the agent nodes from this node pool.
+	// Max value of SwapFileSizeMB should be the size of temporary disk(/dev/sdb). Refer: https://learn.microsoft.com/en-us/azure/virtual-machines/managed-disks-overview#temporary-disk
+	SwapFileSizeMB *int32
+
+	// Sysctl specifies the settings for Linux agent nodes.
+	Sysctls *SysctlConfig
```

Comment:
Instead of having to repeat the struct in this file, have you considered using `*infrav1. SysctlConfig` directly?

Reply:
I think this was a good suggestion. I did not think of reusing `infrav1.LinuxOSConfig` struct, was merely following another pattern. 
Following this suggestion, I refactored this code further more and was able to remove the structs altogether from this file. 

Although this meant I perform some validation at the webhook side.

Reply:
Please take a look at the commit 4523aaee2e44352ce2b413896823206526885b97. 
With 4523aaee2e44352ce2b413896823206526885b97 applied, [azure/services/agentpools/spec.go#L356](https://github.com/nawazkh/cluster-api-provider-azure/blob/4523aaee2e44352ce2b413896823206526885b97/azure/services/agentpools/spec.go#L356) uses `infrav1.LinuxOSConfig`. 

Also note that `NetIpv4IPLocalPortRange` type changes from `type struct` to `type *string` as accepted by Azure API.